gobblin-modules/gobblin-kafka-09/src/test/java/org/apache/gobblin/kafka/KafkaClusterTestBase.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.kafka;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Properties;
+
+import org.I0Itec.zkclient.ZkClient;
+
+import kafka.server.KafkaConfig;
+import kafka.server.KafkaServer;
+import kafka.utils.MockTime;
+import kafka.utils.Time;
+import kafka.utils.ZKStringSerializer$;
+import kafka.zk.EmbeddedZookeeper;
+
+import org.apache.gobblin.test.TestUtils;
+
+public class KafkaClusterTestBase extends KafkaTestBase {
+
+	int clusterCount; 
+	EmbeddedZookeeper _zkServer;
+	String _zkConnectString;
+	ZkClient _zkClient;
+	List<KafkaServer> kafkaBrokerList = new ArrayList<KafkaServer>();
+	List<Integer> kafkaBrokerPortList = new ArrayList<Integer>();
+	
+	public KafkaClusterTestBase(int clusterCount) throws InterruptedException, RuntimeException {
+		super();
+		this.clusterCount = clusterCount;
+	}
+
+	public void startCluster() {
+		// Start Zookeeper.
+		_zkServer = new EmbeddedZookeeper();
+		_zkConnectString = "127.0.0.1:"+_zkServer.port();
+		_zkClient = new ZkClient(_zkConnectString, 30000, 30000, ZKStringSerializer$.MODULE$);
+		// Start Kafka Cluster.
+	    for(int i=0;i<clusterCount;i++) {
+	    	KafkaServer _kafkaServer = createKafkaServer(i,_zkConnectString);
+	    	kafkaBrokerList.add(_kafkaServer);
+	    }
+	}
+
+	public void stopCluster() {
+		Iterator<KafkaServer> iter = kafkaBrokerList.iterator();
+		while(iter.hasNext()){
+			KafkaServer server = iter.next();
+			server.shutdown();
```

Comment:
Please wrap this in try catch, so that if one server has crashed / shutdown prematurely, the exception in shutting it down does not cause others to not close. This can lead to intermittently failing test cases